src/After.tsx

```diff
@@ -21,7 +21,7 @@ class Afterparty extends React.Component<any, any> {
       // save the location so we can render the old screen
       this.setState({
         previousLocation: this.props.location,
-        data: undefined, // unless you want to keep it
+        data: this.state.data
```

Comment:
@jeetiss if you just delete this line, it should preserve the data (sinde setState will merge with the old state).
Could you try this branch https://github.com/jaredpalmer/after.js/pull/130? One of the changes is basically, removing this line.